src/common/common/surface.h

```diff
@@ -317,10 +363,77 @@ void guac_common_surface_clip(guac_common_surface* surface, int x, int y, int w,
 void guac_common_surface_reset_clip(guac_common_surface* surface);
 
 /**
- * Flushes the given surface, drawing any pending operations on the remote
- * display.
+ * Changes the location of the surface relative to its parent layer. If the
+ * surface does not represent a non-default visible layer, then this function
+ * has no effect.
  *
- * @param surface The surface to flush.
+ * @param surface
+ *     The surface to move relative to its parent layer.
+ *
+ * @param x
+ *     The new X coordinate for the upper-left corner of the layer, in pixels.
+ *
+ * @param y
+ *     The new Y coordinate for the upper-left corner of the layer, in pixels.
+ */
+void guac_common_surface_move(guac_common_surface* surface, int x, int y);
+
+/**
+ * Changes the stacking order of the surface relative to other surfaces within
+ * the same parent layer. If the surface does not represent a non-default
+ * visible layer, then this function has no effect.
+ *
+ * @param surface
+ *     The surface to move relative to its parent layer.
+ *
+ * @param z
+ *     The new Z-order for this layer, relative to sibling layers.
+ */
+void guac_common_surface_stack(guac_common_surface* surface, int z);
+
+/**
+ * Changes the parent layer of ths given surface. By default, layers will be
+ * children of the default layer. If the surface does not represent a
+ * non-default visible layer, then this function has no effect.
+ *
+ * @param surface
+ *     The surface to move relative to its parent layer.
```

Comment:
Looks like maybe some bad copypasta here? I don't think this parameter description works here, or in guac_common_surface_set_opacity.